src/main/java/com/palantir/common/streams/MoreFutures.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.common.streams;
+
+import com.google.common.util.concurrent.Futures;
+
+import java.util.concurrent.Future;
+
+class MoreFutures {
+    private MoreFutures() {}
+
+    static <T extends Future<U>, U> T blockOnCompletion(T future) {
```

Comment:
`join` in line with J8 api?

Reply:
no, that has different semantics. I don't want this to throw, whereas join() will throw.